src/components/scenes/SendScene2.tsx

```diff
@@ -0,0 +1,924 @@
+              buttons={{
+                buy: { label: sprintf(s.strings.buy_crypto_modal_buy_action, currencyCode) },
+                exchange: { label: s.strings.buy_crypto_modal_exchange, type: 'primary' },
+                cancel: { label: s.strings.buy_crypto_decline }
+              }}
+            />
+          ))
+          switch (result) {
+            case 'buy':
+              Actions.jump('pluginListBuy', { direction: 'buy' })
+              return
+            case 'exchange':
+              dispatch(selectWalletForExchange(walletId, currencyCode, 'to'))
+              Actions.jump('exchangeScene', {})
+              break
+          }
+        }
+      })
+      .catch(error => console.log(error))
+  }
+
+  const renderAmount = (index: number, spendTarget: EdgeSpendTarget) => {
+    const { publicAddress, nativeAmount } = spendTarget
+    if (publicAddress != null && !hiddenTilesMap.amount) {
+      const title = s.strings.fio_request_amount + (spendInfo.spendTargets.length > 1 ? ` ${(index + 1).toString()}` : '')
+      return (
+        <EditableAmountTile
+          title={title}
+          exchangeRates={exchangeRates}
+          nativeAmount={nativeAmount ?? ''}
+          wallet={coreWallet}
+          currencyCode={currencyCode}
+          exchangeDenomination={cryptoExchangeDenomination}
+          displayDenomination={cryptoDisplayDenomination}
+          lockInputs={lockTilesMap.amount ?? false}
+          // TODO: Handle press
+          onPress={handleFlipInputModal(index, spendTarget)}
+        />
+      )
+    }
+
+    return null
+  }
+
+  const renderAddressAmountPairs = () => {
+    const out: Array<JSX.Element | null> = []
+    for (let i = 0; i < spendInfo.spendTargets.length; i++) {
+      const spendTarget = spendInfo.spendTargets[i]
+      let element
+      if (i < spendInfo.spendTargets.length - 1) {
+        element = renderAddressAmountTile(i, spendTarget)
+        if (element != null) out.push(element)
+      } else {
+        element = renderAddressTile(i, spendTarget)
+        if (element != null) out.push(element)
+        element = renderAmount(i, spendTarget)
+        if (element != null) out.push(element)
+      }
+    }
+    return out
+  }
+
+  const handleWalletPress = useHandler(() => {
+    Airship.show<WalletListResult>(bridge => (
+      <WalletListModal bridge={bridge} headerTitle={s.strings.fio_src_wallet} allowedCurrencyCodes={allowedCurrencyCodes} />
+    ))
+      .then((result: WalletListResult) => {
+        if (result.walletId == null || result.currencyCode == null) {
+          return
+        }
+        setWalletId(result.walletId)
+        const { pluginId: newPluginId } = currencyWallets[result.walletId].currencyInfo
+        if (pluginId !== newPluginId || currencyCode !== result.currencyCode) {
+          setCurrencyCode(result.currencyCode)
+          setSpendInfo({ spendTargets: [{}] })
+        }
+      })
+      .catch(error => console.log(error))
```

Comment:
Why not `.catch(showError)`?

Reply:
copy/paste from original SendScene, but it's an easy fix so I'll update.